### PR TITLE
BasicInput component

### PR DESCRIPTION
**Pull request by coulterpeterson**
## Audience

Add-on Developers | Local Engineers

## Summary

Add new components `BasicInput` (just about the same as InputPasswordToggle, without the password dots or the toggling) and `TableListMultiDisplay` (lets you pass in an array of data that needs to be displayed as rows in a table) to facilitate Add-on development.

### CODE REVIEW

File: src/common/structures/IState.ts [hunk 1]
```diff
@@ -0,0 +1,5 @@
+export default interface IState {
+	addingItem: boolean;
+	initialData: any;
+	unsavedData: any;
+}
```
Comment (flycrum) [inline src/common/structures/IState.ts:5]: Looks like this isn't being imported anywhere and can be removed.

File: src/components/inputs/BasicInput/BasicInput.scss [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+.BasicInput {
+    position: relative;
+    display: block;
+    margin-bottom: 15px;
+
+    input {
+        @include flywheelInput;
+    }
+}
```
Comment (flycrum) [inline src/components/inputs/BasicInput/BasicInput.scss:11]: Not sure if our editor rules are being ignored by editors, but would you change these indentations from spaces to tabs?

File: src/components/inputs/BasicInput/BasicInput.test.tsx [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+it ('renders without crashing', () => {
+	shallow(<BasicInput />);
+});
+
+it ('contains a regular input by default', () => {
+	const component = shallow(<BasicInput />);
+
+	expect(component.exists('input[type="text"]')).toBe(true);
+	expect(component.exists('input[type="password"]')).toBe(false);
+});
```
Comment (flycrum) [inline src/components/inputs/BasicInput/BasicInput.test.tsx:14]: 🙌

File: src/components/inputs/BasicInput/README.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+BasicInput functions just the same as `InputPasswordToggle`, without the password dots or the toggling. So, you know, like a basic input field.
```
Comment (flycrum) [inline src/components/inputs/BasicInput/README.md:1]: Love it 😆

File: src/components/tables/TableListMultiDisplay/README.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
+TableListMultiDisplay lets you pass in an array of data that needs to be displayed as rows in a table. Unlike TableListRepeater, it does not include buttons for the user to add and delete rows. The action button is hidden by default but can be shown with the `noActionButton={false}` prop.
```
Comment (flycrum) [inline src/components/tables/TableListMultiDisplay/README.md:1]: Very nicely documented!

File: src/components/tables/TableListMultiDisplay/TableListMultiDisplay.tsx [hunk 6]
```diff
@@ -0,0 +1,138 @@
+import * as React from "react";
+import IReactComponentProps from "../../../common/structures/IReactComponentProps";
+import classnames from "classnames";
+import { TableList } from "../TableList/TableList";
+import isEqual = require("lodash.isequal");
+import * as styles from "../TableList/TableList.sass";
+import { PrimaryButton } from "../../buttons/PrimaryButton/PrimaryButton";
+import { FunctionGeneric } from "../../../common/structures/Generics";
```
Comment (flycrum) [inline src/components/tables/TableListMultiDisplay/TableListMultiDisplay.tsx:8]: Not sure if it's a config or editor issue, but would you mind changing these from double quotes to single?

File: src/index.ts [hunk 7]
```diff
@@ -1,83 +1,99 @@
...
+export { default as IReactComponentProps } from "./common/structures/IReactComponentProps";
 export {
 	IVirtualTableCellRendererDataArgs,
 	IVirtualTableOnChangeRowDataArgs,
 	IVirtualTableRowRendererDataArgs,
 	VirtualTableCellRenderer,
-	VirtualTableRowRenderer,
-} from './components/modules/VirtualTable/VirtualTable';
+	VirtualTableRowRenderer
+} from "./components/modules/VirtualTable/VirtualTable";
```
Comment (flycrum) [inline src/index.ts:99]: Same thing here, would mind changing these to single quotes? Not sure how it got all funky 🤔

**Comment by coulterpeterson** [comment]
@flycrum thanks, changes made! As I mentioned in PR #98 I believe I accidentally merged my two branches before making this PR, so this can be the one PR to rule them all if you like :) Let me know if you spot any other needed changes!

(Note: my VS Code is set to replace tabs with 4 spaces, however I overrode this in that `BasicInput.scss` file. If that's needed on all files I've touched let me know)

**Pull request merged**